extensions/nodes/nodes.go

```diff
@@ -0,0 +1,114 @@
+package nodes
+
+import (
+	"github.com/rancher/norman/types"
+	v3 "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	"github.com/rancher/shepherd/clients/rancher"
+	"github.com/rancher/shepherd/extensions/kubeapi/secrets"
+	corev1 "k8s.io/api/core/v1"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	initNodeLabelKey         = "rke.cattle.io/init-node"
+	initNodeLabelValue       = "true"
+	local                    = "local"
+	machinePlanSecretType    = "rke.cattle.io/machine-plan"
+	machineNameSteveLabel    = "rke.cattle.io/machine-name"
+	node                     = "node"
+)
+
+// GetInitNode accepts a client and clusterID and returns the init node object
+func GetInitNode(client *rancher.Client, clusterID string) (*v3.Node, error) {
+	logrus.Info("Retrieving list of secrets...")
+	secretList, err := secrets.ListSecrets(client, local, fleetNamespace, metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	machinePlans := []corev1.Secret{}
+
+	logrus.Info("Identifying machine-plan secrets...")
+	for _, secret := range secretList.Items {
+		if secret.Type == machinePlanSecretType {
+			logrus.Infof("Found machine-plan secret: %s", secret.Name)
+			machinePlans = append(machinePlans, secret)
+		}
+	}
+
+	initNodeMachineName := ""
+	
+	logrus.Info("Identifying init node machine plan...")
+	for _, machinePlan := range machinePlans {
+		for key, val := range machinePlan.ObjectMeta.Labels {
+			if key == initNodeLabelKey && val == initNodeLabelValue {
+				logrus.Infof("Found init node machine plan: %s", machinePlan.Name)
+				initNodeMachineName = machinePlan.Labels[machineNameSteveLabel]
+			}
+		}
+	}
```

Comment:
A question here, does it need to create nested for loops? It can check map keys, something like this;

```suggestion
	for _, machinePlan := range machinePlans {
	        v, ok := machinePlan.ObjectMeta.Labels[initNodeLabelKey]
	        if ok && v ==  initNodeLabelValue {
	             		logrus.Infof("Found init node machine plan: %s", machinePlan.Name)
				initNodeMachineName = machinePlan.Labels[machineNameSteveLabel]
		}
	}
```

Reply:
great suggestion! thank you Ugur! i've implemented this